tests/ttsim/mettsim/orc_hunting_bounty/orc_hunting_bounty.py

```
@@ -1,23 +1,48 @@
-"""Orc Hunting Bounty"""
+from __future__ import annotations
 
-from ttsim import DictTTSIMParam, policy_function
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+from ttsim import params_function, policy_function
+
+if TYPE_CHECKING:
+    from ttsim.typing import RawParamsRequiringConversion
+
+
+@dataclass(frozen=True)
+class BountyPerLargeOrcs:
+    noble_hunter: float
+    peasant_hunter: float
+
+
+@dataclass(frozen=True)
+class BountyPerOrcs:
+    small_orc: int
+    large_orc: BountyPerLargeOrcs
+
+
+@params_function()
```

I am a bit surprised that there is no difference between functions that process params
(like this one) and functions that return a scalar param only. The former will never be
a target, the latter may be. Should we enforce this?

---

Now I am confused what difference there could possibly be? All functions we have so far that return a scalar for sure will become a `params_function`. Sure, you may want to request them as outputs, but why would you want to have an entire column filled with a single value? The only reason might be when it changes over time and at some point starts depending on the data. But that will just continue to work.

---

My point is that `bounty_per_orc` returns a `BountyPerOrc` object, while other `params_function` may only return a scalar. It makes sense to request the latter as a target, but I'm not sure about the former. If we allow for that, it definitely changes how we define GETTSIM's output (e.g. type hint is wrong).

---

I was trying to say that scalars won't make much sense, either, except for special cases. 

This said, for tree output, anything goes. In case of df output, we should throw an error when requesting dict conversion. See [ii](https://github.com/iza-institute-of-labor-economics/gettsim/pull/897#issue-3041987819) under that to-do; that will still need to be settled.